src/util/sharing_map.h

```diff
@@ -39,6 +39,66 @@ Author: Daniel Poetzl
   CV typename sharing_mapt<keyT, valueT, hashT, predT>::ST \
   sharing_mapt<keyT, valueT, hashT, predT>
 
+/// A map implemented as a tree where subtrees can be shared between different
+/// maps.
+///
+/// The map is implemented as a fixed-height n-ary trie. The height H and the
+/// maximum number of children per inner node S are determined by the two
+/// configuration parameters `bits` and `chunks` in sharing_map.h. It holds
+/// that H = `bits` / `chunks` and S = 2 ^ `chunks`.
+///
+/// When inserting a key-value pair into the map, first the hash of its key is
+/// computed. The `bits` number of lower order bits of the hash are deemed
+/// significant, and are grouped into `bits` / `chunk` chunks). The hash is then
+/// treated as a string (with each chunk representing a character) for the
+/// purposes of determining the position of the key-value pair in the trie. The
+/// actual key-value pairs are stored in the leaf nodes. Collisions (i.e., two
+/// different keys yield the same "string"), are handled by chaining the
+/// corresponding key-value pairs in a `std::list`.
```

Comment:
Would a `std::forward_list` be sufficient?

Reply:
Yes, this is fixed in an upcoming PR.